### PR TITLE
Adjust code samples to ember-cli

### LINKED ISSUE #3155: Update docs to ember-cli style

### PR DESCRIPTION
**Pull request by Turbo87**
I'm adjusting the samples to match the [blueprints](https://github.com/ember-cli/ember-cli/tree/master/blueprints) of the `ember-cli` tool as discussed in #3139.
Closes #3155

### CODE REVIEW

File: packages/ember-data/lib/transforms/base.js [hunk 1]
```diff
@@ -8,8 +8,10 @@
   Example
 
   ```javascript
+  import DS from 'ember-data';
```
Comment (wecc) [inline packages/ember-data/lib/transforms/base.js]: Add `// app/transforms/temperature.js` here?

Comment (Turbo87) [inline packages/ember-data/lib/transforms/base.js]: yep, good idea. done.

File: packages/ember-data/lib/adapters/rest-adapter.js [hunk 2]
```diff
@@ -99,7 +102,7 @@ import BuildURLMixin from "ember-data/adapters/build-url-mixin";
   property on the adapter:
 
   ```js
-  App.ApplicationAdapter = DS.RESTAdapter.extend({
+  export default DS.RESTAdapter.extend({
```
Comment (sly7-7) [inline packages/ember-data/lib/adapters/rest-adapter.js]: Maybe it's worth it to add the file name here too ?

File: packages/ember-data/lib/adapters/build-url-mixin.js [hunk 3]
```diff
@@ -251,7 +251,7 @@ export default Ember.Mixin.create({
     endpoint of "/line_items/".
 
     ```js
-    App.ApplicationAdapter = DS.RESTAdapter.extend({
+    export default DS.RESTAdapter.extend({
```
Comment (bmac) [inline packages/ember-data/lib/adapters/build-url-mixin.js]: `// app/adapters/application.js`

**Comment by Turbo87** [comment]
@bmac @sly7-7 done

**Comment by sly7-7** [comment]
@Turbo87 @bmac Maybe the header `//app/adapters/application.js`could added in every snippet where we extend the adapter ? Or maybe it's enough to have it in one example ?

**Comment by bmac** [comment]
I think it should be in every example. Most people just look at the one api they are in erected in instead of reading the whole page.

**Comment by Turbo87** [comment]
@sly7-7 I felt like it might be too much visual clutter if we include it everywhere. It might also make people believe that this is only valid for the `application` adapter, which is obviously not true. I haven't found a good rule yet on when to include it, but I think if the example explains more then just a simple method or property (e.g. an interaction with a model) it should definitely be included.

**Comment by sly7-7** [comment]
@Turbo87 It seems like you added it to all the snippets for serializers/embedded mixin, so for consistency, either we only have one per file, or one per snippets. 
That beeing said, when I played with ember-cli and ember-data I remember me trying to find in which file the snippets should be included, so my opinion would be the same as @bmac. 
Maybe we could add/replace an example with a per-type adapter, so that it would help to figure out it's not only valid for `application` adapter ?

**Comment by bmac** [comment]
We should look into the helper that the guides use to make their pretty file names and see if it needs to be added to the ember/website repo.

**Comment by igorT** [comment]
paging @rwjblue

**Comment by bmac** [comment]
Looks like the code is already on the website repo: https://github.com/emberjs/website/commit/3bd5ef62426cf7b2a9f1cccb4bc00a38de8e0674

**Comment by bmac** [comment]
Looks like I spoke too soon the helper works but looks ugly. I've opened a pr to fix this: https://github.com/emberjs/website/pull/2192.

@Turbo87 Do you mind updating this pr to remove the all of the filename comments and replace the `js` after the 3 backticks with the filename. 
```
```js
// app/adapters/application.js
code...
```
->
```
```app/adapters/application.js
code...
```

**Comment by Turbo87** [comment]
@bmac @sly7-7 done

File: packages/ember-data/lib/system/model/attributes.js [hunk 4]
```diff
@@ -24,14 +24,16 @@ Model.reopenClass({
     Example
 
     ```javascript
```
Comment (sly7-7) [inline packages/ember-data/lib/system/model/attributes.js]: Maybe `app/models/person.js here` ?

File: packages/ember-data/lib/system/model/attributes.js [hunk 5]
```diff
@@ -24,14 +24,16 @@ Model.reopenClass({
...
 
-    App.Person = DS.Model.extend({
+    var Person = DS.Model.extend({
       firstName: attr('string'),
       lastName: attr('string'),
       birthday: attr('date')
     });
 
-    var attributes = Ember.get(App.Person, 'attributes')
+    var attributes = Ember.get(Person, 'attributes')
```
Comment (sly7-7) [inline packages/ember-data/lib/system/model/attributes.js]: not sure what to do in those cases, where we consume the class.
Either use the `app/models/person.js` header, either replace the class declaration by an import statement like `import Person from app/models/person` ?

File: packages/ember-data/lib/system/model/attributes.js [hunk 6]
```diff
@@ -24,14 +24,16 @@ Model.reopenClass({
     Example
 
     ```javascript
+    import Ember from 'ember';
+    import DS from 'ember-data';
 
-    App.Person = DS.Model.extend({
+    var Person = DS.Model.extend({
```
Comment (Turbo87) [inline packages/ember-data/lib/system/model/attributes.js]: @sly7-7 the problem is the `Ember.get()` call below. I'm not sure how this is supposed to work with ember-cli now. Is there something similar that could replace it?

File: packages/ember-data/lib/system/model/attributes.js [hunk 7]
```diff
@@ -71,13 +73,17 @@ Model.reopenClass({
...
+    import DS from 'ember-data';
+
+    export default DS.Model.extend({
       firstName: attr(),
       lastName: attr('string'),
       birthday: attr('date')
     });
+    ```
 
+    ```javascript
```
Comment (sly7-7) [inline packages/ember-data/lib/system/model/attributes.js]: maybe adding the import statement

File: packages/ember-data/lib/system/model/attributes.js [hunk 8]
```diff
@@ -71,13 +73,17 @@ Model.reopenClass({
...
     var transformedAttributes = Ember.get(App.Person, 'transformedAttributes')
```
Comment (sly7-7) [inline packages/ember-data/lib/system/model/attributes.js]: and in that case I think it should be `Ember.get(Person, 'transformedAttributes')`

File: packages/ember-data/lib/system/store.js [hunk 9]
```diff
@@ -1669,11 +1683,21 @@ Store = Service.extend({
...
+    export default DS.ActiveModelSerializer;
+    ```
+
+    ```app/serializers/post.js
+    import DS from 'ember-data';
+
+    export default DS.JSONSerializer;
+    ```
+
     ```js
```
Comment (sly7-7) [inline packages/ember-data/lib/system/store.js]: Maybe replace the serializers declarations by import statements

File: packages/ember-data/lib/system/model/attributes.js (hunk 6 above)
Comment (sly7-7) [inline packages/ember-data/lib/system/model/attributes.js]: I think you could still do 
`import Ember from 'ember'` 
then call the `Ember.get()`
@bmac c/d ?

File: packages/ember-data/lib/system/store.js (hunk 9 above)
Comment (Turbo87) [inline packages/ember-data/lib/system/store.js]: hmm, not sure what you mean?

File: packages/ember-data/lib/system/model/attributes.js (hunk 5 above)
Comment (Turbo87) [inline packages/ember-data/lib/system/model/attributes.js]: done

File: packages/ember-data/lib/system/store.js (hunk 9 above)
Comment (sly7-7) [inline packages/ember-data/lib/system/store.js]: I'm sorry, we don't use them here, it's all good

**Comment by sly7-7** [comment]
Well, this probably needs someone else eyes, but LGTM
Thanks a lot @Turbo87 awesome work !! :heart:

File: README.md [hunk 10]
```diff
@@ -52,49 +52,48 @@ controllers in your app.
...
+
 Here's what your model definition would look like if you're using
-globals (that is, not something like  or ember-cli):
+ES6 modules (via ember-cli):
 
 ```js
-var attr = DS.attr;
-var hasMany = DS.hasMany;
-var belongsTo = DS.belongsTo;
+// app/models/blog-post.js
```
Comment (bmac) [inline README.md]: Lets move this to the header.

File: packages/ember-data/lib/serializers/json-serializer.js [hunk 11]
```diff
@@ -682,7 +714,7 @@ export default Serializer.extend({
         }
       }
     });
-   ```
+    ```
```
Comment (bmac) [inline packages/ember-data/lib/serializers/json-serializer.js]: Is this a typo?

Comment (Turbo87) [inline packages/ember-data/lib/serializers/json-serializer.js]: unfortunately the indentation in the comments is not always consistent and in this case it used the wrong indentation previously. it is now aligned properly with the rest of the comment.

File: README.md (hunk 10 above)
Comment (bmac) [inline README.md]: Ah nevermind this is the `README.md`.

File: packages/ember-data/lib/serializers/json-serializer.js (hunk 11 above)
Comment (bmac) [inline packages/ember-data/lib/serializers/json-serializer.js]: Thanks.

File: README.md (hunk 10 above)
Comment (Turbo87) [inline README.md]: do you mean <code>```app/models/blog-post.js</code>? that won't work here since it will be rendered by GitHub and not the website.

**Comment by bmac** [comment]
Looks good @Turbo87 :tada:

**Comment by Turbo87** [comment]
cool, thanks for merging!

**Pull request merged**